src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs

```diff
@@ -346,23 +346,33 @@ public Task SendStatusAsync(int statusCode)
         {
             ApplyResponseHeaders(Constants.Status200Ok);
 
-            string physicalPath = _fileInfo.PhysicalPath;
-            var sendFile = _context.Features.Get<IHttpSendFileFeature>();
-            if (sendFile != null && !string.IsNullOrEmpty(physicalPath))
-            {
-                await sendFile.SendFileAsync(physicalPath, 0, _length, _context.RequestAborted);
-                return;
-            }
-
-            Stream readStream = _fileInfo.CreateReadStream();
             try
             {
-                // Larger StreamCopyBufferSize is required because in case of FileStream readStream isn't going to be buffering
-                await StreamCopyOperation.CopyToAsync(readStream, _response.Body, _length, StreamCopyBufferSize, _context.RequestAborted);
+                string physicalPath = _fileInfo.PhysicalPath;
+                var sendFile = _context.Features.Get<IHttpSendFileFeature>();
+                if (sendFile != null && !string.IsNullOrEmpty(physicalPath))
+                {
+                    await sendFile.SendFileAsync(physicalPath, 0, _length, _context.RequestAborted);
```

Comment:
hmm, maybe we don't need it for SendFileAsync, but we do want it for the manual copy below. Files may be large and we don't want to keep reading from disk if the writes are failing.